### DISCUSSION TITLE
Having a certificate problem

### DISCUSSION THREAD

**Discussion by direwolf2020** [rfennell/AzurePipelines#1481] [General]
When I try to run the release note generator, I am receiving a certificate error:

Starting: Generate Release Notes based on Release Comparison API
==============================================================================
Task         : Generate Release Notes (Crossplatform)
Description  : Generates a release notes file in a format of your choice from the build or release history
Version      : 4.4.1
Author       : Black Marble
Help         : Version: 4.4.1. [More Information](https://github.com/rfennell/AzurePipelines/wiki/GenerateReleaseNotes---Node-based-Cross-Platform-Task/)
==============================================================================
Creating Azure DevOps API connections for https://xxx.xxx.xxx/Apps/ with 'allowRetries' set to 'true' and 'maxRetries' count to '20'
Creating the credential handler from the OAUTH token
##[error]Unhandled: unable to get local issuer certificate
##[error]Error: unable to get local issuer certificate
    at TLSSocket.onConnectSecure (node:_tls_wrap:1532:34)
    at TLSSocket.emit (node:events:527:28)
    at TLSSocket._finishInit (node:_tls_wrap:946:8)
    at TLSWrap.ssl.onhandshakedone (node:_tls_wrap:727:12)
Finishing: Generate Release Notes based on Release Comparison API

I believe I the cert assigned everywhere....any other places I should look?

Thanks!

**Comment by rfennell** [comment]
All the URL handling for the API is automatically managed using Azure DevOps environment variables, so there are no settings to alter related to URLs or Certs,

Are you trying to use this with an on premises Azure DevOps Server, if so which version?

**Comment by direwolf2020** [comment]
It is an on premises install.

We are running Azure DevOps Server 2020 Update 1.1 (18.181.31626.1).

I've tried everything that I can think of to get it to recognize the
certificate!

**Reply by rfennell** [reply]
Have you looked at

- #801 - a workaround using a PAT
- #842 - incorrectly assign self signed cert

**Comment by direwolf2020** [comment]
I'm really new to setting this up, so I didn't understand what to do in
some of  the suggestions.

I set the NODE_TLS_REJECT_UNAUTHORIZED = 0, and now I am receiving this
error message:


##[error]Error: ﻿{"$id":"1","innerException":null,"message":"TF400813:
Resource not available for anonymous access. Client authentication
required.","typeName":"Microsoft.TeamFoundation.Framework.Server.UnauthorizedRequestException,
Microsoft.TeamFoundation.Framework.Server","typeKey":"UnauthorizedRequestException","errorCode":0,"eventId":3000}

**Reply by rfennell** [reply]
Can you run the build with the variable

 ```
system.debug=true
```

And send the detailed run log so I can try to see what is going on. I need to see when the error is occurring so we can link it back to the code.

Also what version of azure devops server are you using, I assume it is not azure devops services

**Comment by direwolf2020** [comment]
I ran the release with the debug set, but it returned the same messages.

Should this task be run on the release pipeline, or just the pipeline
itself?

**Reply by rfennell** [reply]
The task should work a YAML Builds and both classic Build and Release, is the error the first thing in the task's log file?

Did 
- #801 - a workaround using a PAT
- #842 - incorrectly assign self signed cert

help?
